velox/exec/tests/LimitTest.cpp

```diff
@@ -63,3 +63,42 @@ TEST_F(LimitTest, basic) {
 
   assertQueryReturnsEmptyResult(makePlan(12'345, 10));
 }
+
+TEST_F(LimitTest, limitOverLocalExchange) {
+  auto data = makeRowVector(
+      {makeFlatVector<int32_t>(1'000, [](auto row) { return row; })});
+
+  auto file = TempFilePath::create();
+  writeToFile(file->path, {data});
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId scanNodeId;
+
+  CursorParameters params;
+  params.planNode =
+      PlanBuilder(planNodeIdGenerator)
+          .localPartition(
+              {},
+              {PlanBuilder(planNodeIdGenerator)
+                   .tableScan(
+                       std::dynamic_pointer_cast<const RowType>(data->type()))
+                   .capturePlanNodeId(scanNodeId)
+                   .planNode()})
+          .limit(0, 20, true)
+          .planNode();
+
+  TaskCursor cursor(params);
+  addSplit(cursor.task().get(), scanNodeId, makeHiveSplit(file->path));
+
+  int32_t numRead = 0;
+  while (cursor.moveNext()) {
+    auto vector = cursor.current();
+    numRead += vector->size();
+  }
+
+  // Do not send no-more-splits message. Expect the task to finish without
```

Comment:
I feel we should have a timeout here that throws an error.

Reply:
It would be nice for tests to have a generic timeout functionality, but looks like GoggleTest doesn't offer that. Some folks suggested using timeout support in ctest.

https://stackoverflow.com/questions/16942154/how-to-add-timeout-to-test-when-using-google-testing-framework

https://cmake.org/cmake/help/v3.8/prop_test/TIMEOUT.html

Should we look into these options?

Reply:
I tried setting `set_tests_properties(velox_exec_test PROPERTIES TIMEOUT 1)` and without `params.numResultDrivers = 1;`. Strangely it did not timeout. 
Probably adding a timeout can be a separate PR.